homeassistant/components/renault/config_flow.py

```diff
@@ -90,3 +91,54 @@ async def async_step_kamereon(
                 {vol.Required(CONF_KAMEREON_ACCOUNT_ID): vol.In(accounts)}
             ),
         )
+
+    async def async_step_reauth(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+        if not self._existing_entry:
+            if TYPE_CHECKING:
+                assert user_input
+            await self.async_set_unique_id(user_input[CONF_KAMEREON_ACCOUNT_ID])
+            self._existing_entry = user_input.copy()
+            user_input = None
+
+        if user_input is None:
+            return self._show_reauth_form()
+
+        self.renault_hub = RenaultHub(self.hass, self._existing_entry[CONF_LOCALE])
+        if not await self.renault_hub.attempt_login(
+            self._existing_entry[CONF_USERNAME], user_input[CONF_PASSWORD]
+        ):
+            return self._show_reauth_form({"base": "invalid_credentials"})
+
+        data = self._existing_entry.copy()
+        data[CONF_PASSWORD] = user_input[CONF_PASSWORD]
+        existing_entry = await self.async_set_unique_id(
+            self._existing_entry[CONF_KAMEREON_ACCOUNT_ID]
+        )
+        if TYPE_CHECKING:
+            assert existing_entry
+        self.hass.config_entries.async_update_entry(existing_entry, data=data)
+        await self.hass.config_entries.async_reload(existing_entry.entry_id)
+        return self.async_abort(reason="reauth_successful")
+
+    def _show_reauth_form(self, errors: dict[str, Any] | None = None) -> FlowResult:
+        """Show the API keys form."""
+        if TYPE_CHECKING:
+            assert self._existing_entry
+        return self.async_show_form(
+            step_id="reauth",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_USERNAME,
+                        description={
+                            "suggested_value": self._existing_entry[CONF_USERNAME]
+                        },
+                    ): str,
```

Comment:
Not sure if there is a better way to show the original username. For now it's editable but ignored.